src/cascadia/TerminalApp/TabBase.cpp

```diff
@@ -252,4 +254,289 @@ namespace winrt::TerminalApp::implementation
         });
     }
 
+    std::optional<winrt::Windows::UI::Color> TabBase::GetTabColor()
+    {
+        return std::nullopt;
+    }
+
+    void TabBase::ThemeColor(const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& focused,
+                             const winrt::Microsoft::Terminal::Settings::Model::ThemeColor& unfocused,
+                             const til::color& tabRowColor)
+    {
+        _themeColor = focused;
+        _unfocusedThemeColor = unfocused;
+        _tabRowColor = tabRowColor;
+        _RecalculateAndApplyTabColor();
+    }
+
+    // Method Description:
+    // - This function dispatches a function to the UI thread to recalculate
+    //   what this tab's current background color should be. If a color is set,
+    //   it will apply the given color to the tab's background. Otherwise, it
+    //   will clear the tab's background color.
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TabBase::_RecalculateAndApplyTabColor()
+    {
+        auto weakThis{ get_weak() };
+
+        TabViewItem().Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [weakThis]() {
+            auto ptrTab = weakThis.get();
+            if (!ptrTab)
+                return;
+
+            auto tab{ ptrTab };
+
+            // GetTabColor will return the color set by the color picker, or the
+            // color specified in the profile. If neither of those were set,
+            // then look to _themeColor to see if there's a value there.
+            // Otherwise, clear our color, falling back to the TabView defaults.
+            auto currentColor = tab->GetTabColor();
+            if (currentColor.has_value())
+            {
+                tab->_ApplyTabColor(currentColor.value());
+            }
+            else if (tab->_themeColor != nullptr)
+            {
+                // Safely get the active control's brush.
+                Media::Brush terminalBrush{ tab->_BackgroundBrush() };
+
+                if (const auto themeBrush{ tab->_themeColor.Evaluate(Application::Current().Resources(), terminalBrush, false) })
+                {
+                    // ThemeColor.Evaluate will get us a Brush (because the
+                    // TermControl could have an acrylic BG, for example). Take
+                    // that brush, and get the color out of it. We don't really
+                    // want to have the tab items themselves be acrylic.
+                    tab->_ApplyTabColor(til::color{ ThemeColor::ColorFromBrush(themeBrush) });
+                }
+                else
+                {
+                    tab->_ClearTabBackgroundColor();
+                }
+            }
+            else
+            {
+                tab->_ClearTabBackgroundColor();
+            }

```

Comment:
Uh, because this is the case where there was no tabColor set (either in the control itself (from VT or profile.tabColor), or by the tab color picker). There's no custom color here, so just, get rid of it. (this is from the original color picker code, basically)